Antidote/CallsManager.m

```diff
@@ -0,0 +1,331 @@
+{
+    // grab paused call from RBQ index and resume call.
+}
+
+#pragma mark - DialingCallViewController Delegate
+
+- (void)dialingCallDeclineButtonPressed:(DialingCallViewController *)controller
+{
+    [self.manager sendCallControl:OCTToxAVCallControlCancel
+                           toCall:self.currentCall
+                            error:nil];
+}
+
+#pragma mark - RingingCallViewController Delegate
+
+- (void)ringingCallAnswerButtonPressed:(RingingCallViewController *)controller
+{
+    [self.manager answerCall:self.currentCall
+                 enableAudio:YES
+                 enableVideo:NO error:nil];
+}
+
+- (void)ringingCallDeclineButtonPressed:(RingingCallViewController *)controller
+{
+    [self.manager sendCallControl:OCTToxAVCallControlCancel
+                           toCall:self.currentCall
+                            error:nil];
+}
+
+#pragma mark - Private
+
+- (void)notifyOfIncomingCall:(OCTCall *)call
+{
+    if (! [self.currentCallViewController isKindOfClass:[ActiveCallViewController class]]) {
+        // For now we reject call if we are in a dialing or ringing state
+        [self.manager sendCallControl:OCTToxAVCallControlCancel toCall:call error:nil];
+        return;
+    }
+
+    self.pendingIncomingCall = call;
+
+    OCTFriend *friend = [self.pendingIncomingCall.chat.friends firstObject];
+
+    ActiveCallViewController *activeVC = (ActiveCallViewController *)self.currentCallViewController;
+    activeVC.incomingCallCallerName = friend.nickname;
+    activeVC.createIncomingCallView = YES;
+}
+
+- (void)dealloc
+{
+    [self.currentCallViewController dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (AbstractCallViewController *)viewControllerForCall:(OCTCall *)call
+{
+    AbstractCallViewController *viewController;
+
+    OCTFriend *friend = [call.chat.friends firstObject];
+
+    switch (call.status) {
+        case OCTCallStatusActive:
+            viewController = [ActiveCallViewController new];
+            break;
+        case OCTCallStatusDialing:
+            viewController = [DialingCallViewController new];
+            break;
+        case OCTCallStatusRinging:
+            viewController = [RingingCallViewController new];
+            break;
+        case OCTCallStatusPaused:
+            NSAssert(NO, @"We should not be here yet. Not yet implemented");
+            break;
+    }
+
+    viewController.nickname = friend.nickname;
+    viewController.modalInPopover = YES;
+    viewController.modalPresentationStyle = UIModalPresentationOverCurrentContext;
+    viewController.delegate = self;
+
+    return viewController;
+}
+
+- (void)dismissAndSwitchViewControllerForCall:(OCTCall *)call
+{
+    AbstractCallViewController *abstractVC = [self viewControllerForCall:call];
+
+    TabBarViewController *tabBarVC = (TabBarViewController *)[AppContext sharedContext].tabBarController;
+
+    if ([tabBarVC presentedViewController]) {
```

Comment:
Hm... I'm not happy with that. There can be some other controller presented, not necessary calls. I think itwould be better to check if there is `currentCallViewController` and dismiss it.

Or you can use NavigationController with one controller is stack. When you need to change controllers just replace `viewControllers` property with new controller